TFT/src/User/API/Vfs/vfs.h

```diff
@@ -18,6 +19,7 @@ typedef enum {
 typedef struct
 {
 	TCHAR title[MAX_PATH_LEN]; // max path depth
+	TCHAR cur_file[255]; // current file selected for print
```

Comment:
Why did you add `cur_file` when `title` already exists for the same purpose? This is not a good use of already limited RAM.

Reply:
Because I cannot change title to long filename as it is used in multiple places for different purposes. I also first started to use only an index of the current file and use it to get the long filename based on that index (from infoFile.Longfile) but that way I couldn't store the filename if printing from other USB host (ex Octoprint) that uses onboard SD.

Reply:
There will be no info about the file list in Onboard SD if the print is started by another host.  In **parseACK.c** `cur_file` is assigned same value as `title`. how is it different?

Reply:
There will be info about the file list in Onboard SD if the print is started by another host. Just test it, I did. In fact if you start the print from another host than TFT and you print from the onboard SD, you will have the Print Menu active when print is started, just as you would have if you start the print from the TFT.
In **parseACK.c** `cur_file` is assigned almost the same value as `title` ONLY IF `cur_file` is empty. It's empty because the file selected for printing was not done on the TFT but on another host. M33 is useless once you start the print from another host, hence I cannot retrieve the long filename in that case.